File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0,profit = 0, low = prices[0];
        for(int i = 1;i< prices.size();i++){
            if(prices[i]<low){
                profit = 0;
                low = prices[i];
            }else{
                profit = prices[i]-low;
            }
            max_profit = std::max(profit,max_profit);
        }
        return max_profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit,max_profit,low = 0,0,sys.maxsize
        for price in prices:
            if price < low:
                low = price
                profit = 0
            else:
                profit = price - low
            max_profit = max(profit,max_profit)
        return max_profit